### PR TITLE
Housekeeping for documentation

### PR DESCRIPTION
**Pull request by cmouse**

### CODE REVIEW

File: source/admin_manual/mda.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+:ref:`MTA <mta>` and delivers it to a real or
+virtual mailbox.
+
+Common choices include:
+
+-  `maildrop <https://maildrop.cc>`_
+
+-  `procmail <http://www.procmail.org/>`_ (appears to be unmaintained)
+
+-  :ref:`Dovecot's \`deliver\` LDA <lda>`
```
Comment (sirainen) [inline source/admin_manual/mda.rst:22]: it's called dovecot-lda nowadays

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/admin_manual/mailbox_formats/maildir.rst [hunk 2]
```diff
@@ -189,7 +189,6 @@ Filename Examples
 |                                                                                                   | that folder's ``dovecot-keywords`` file.                 |
 +---------------------------------------------------------------------------------------------------+----------------------------------------------------------+
 
-.. _`IMAP INTERNALDATE, RFC 3501 [2.3.3]`: https://tools.ietf.org/html/rfc3501#section-2.3.3
 .. _`index files`: https://wiki.dovecot.org/IndexFiles
```
Comment (fleischie) [inline source/admin_manual/mailbox_formats/maildir.rst:192]: This is already migrated to the current docs:
https://doc.dovecot.org/developer_manual/design/indexes/mail_index_api/

File: source/admin_manual/sasl.rst [hunk 3]
```diff
@@ -8,7 +8,7 @@ SASL stands for "Simple Authentication and Security Layer". SASL itself is
 nothing more than a list of requirements for
 :ref:`authentication-authentication_mechanisms` and protocols to be
 SASL-compatible as described in
-`RFC 4422 <http://tools.ietf.org/html/rfc4422>`_. IMAP, POP3, SMTP, and
+:rfc:`4422`. IMAP, POP3, SMTP, and
 ManageSieve protocols all have support for SASL.
```
Comment (fleischie) [inline source/admin_manual/sasl.rst:12]: After removing such a long part of the line you could re-flow the text, i.e. join the lines as far as possible, respecting the line-length.

File: source/admin_manual/submission_server.rst [hunk 4]
```diff
@@ -5,22 +5,20 @@ Submission Server
...
 the need to configure the MTA for :ref:`SASL authentication <sasl>`. More SMTP
-capabilities like `CHUNKING <https://tools.ietf.org/html/rfc3030>`_ and `SIZE
-<https://tools.ietf.org/html/rfc1870>`_ are supported, without requiring the
-backend MTA supporting these extensions. Other capabilities like `8BITMIME
-<https://tools.ietf.org/html/rfc6152>`_ and `DSN
-<https://tools.ietf.org/html/rfc3461>`_ currently require support from the
+capabilities like CHUNKING :rfc:`3030` and SIZE :rfc:`1870` are supported, without requiring the
+backend MTA supporting these extensions. Other capabilities like 8BITMIME :rfc:`6152` and
+DS :rfc:`3461` currently require support from the
 backend/relay MTA.
```
Comment (fleischie) [inline source/admin_manual/submission_server.rst:16]: Re-flow text. The paragraph looks visually unraveled now with the removed links.

File: source/admin_manual/submission_server.rst [hunk 5]
```diff
@@ -59,19 +57,19 @@ Features
...
- * `SIZE <https://tools.ietf.org/html/rfc1870>`_
- * `STARTTLS <https://tools.ietf.org/html/rfc3207>`_
- * `VRFY <https://tools.ietf.org/html/rfc5321>`_
- * `XCLIENT <http://www.postfix.org/XCLIENT_README.html>`_
+ * ENHANCEDSTATUSCODES :rfc:`2034`
+ * PIPELINING :rfc:`2920`
+ * SIZE :rfc:`1870`
+ * STARTTLS :rfc:`3207`
+ * VRFY :rfc:`5321`
+ * XCLIENT <http://www.postfix.org/XCLIENT_README.html>
```
Comment (fleischie) [inline source/admin_manual/submission_server.rst]: If I understand the patch correctly you want to display the link behind the feature. Then I would remove the `<` and `>` chars, that are put verbatim into the text now.

File: source/configuration_manual/acl.rst [hunk 6]
```diff
@@ -242,7 +242,7 @@ ID       Type                Description
 ======== =============== ======================================================================================================================================================================================
 
 The ACLs are compatible with
-`RFC 4314 <https://tools.ietf.org/html/rfc4314>`_ (IMAP ACL extension).
+:rfc:`4314` (IMAP ACL extension).
```
Comment (fleischie) [inline source/configuration_manual/acl.rst:245]: This could be one line now.

File: source/configuration_manual/imap_compress.rst [hunk 7]
```diff
@@ -5,7 +5,7 @@ IMAP Compression
 ================
 
 Dovecot supports the 
-`IMAP COMPRESS (RFC 4978) <https://tools.ietf.org/html/rfc4978>`_ extension.
+IMAP COMPRESS (:rfc:`4978`) extension.
```
Comment (fleischie) [inline source/configuration_manual/imap_compress.rst:8]: Can be one line I guess.

**Review by fleischie — CHANGES_REQUESTED** [review]
Cool simplification of the RFC links.

I have additional non-commit/non-change specific comments:

- I did not inspect each individual link change, I glanced over them and inspected some at random.
- Is there a reason you did not port the MDA sub-pages from the wiki?
- The RFC links have an additional `<strong>` tag around them. Is this intended?
- Should we also add a RFC-draft tag? I saw some manual links to RFC drafts.

**Comment by cmouse** [comment]
the additional strong is added by sphinx core code.

**Comment by cmouse** [comment]
> Is there a reason you did not port the MDA sub-pages from the wiki?
LDA page is already ported, maildrop has an actual web page, so I decided to link there.

>The RFC links have an additional <strong> tag around them. Is this intended?
The additional strong is added by sphinx core code.

>Should we also add a RFC-draft tag? I saw some manual links to RFC drafts.
Not that many though.

File: source/admin_manual/submission_server.rst (hunk 5 above)
Comment (cmouse) [reply source/admin_manual/submission_server.rst]: was not really the intention, but i'll do something about it.

File: source/admin_manual/mailbox_formats/mbox.rst [hunk 8]
```diff
@@ -171,7 +171,7 @@ metadata. It does sanity checks for them, so the headers may also be modified
 or removed completely. None of these headers are sent to IMAP/POP3 clients
 when they read the mail.
 
-**The MTA, MDA or LDA should strip all these headers case-insensitively before
+**The :ref:`MTA <mta>`, :ref:`MDA <mda>` or :ref:`LDA <lda>` should strip all these headers case-insensitively before
```
Comment (fleischie) [inline source/admin_manual/mailbox_formats/mbox.rst:174]: This should be in the commit two prior (Re: Adding links to MTA, MDA, and LDA).

**Review by fleischie — CHANGES_REQUESTED** [review]

**Review by fleischie — APPROVED** [review]

**Pull request merged**